53.md

```diff
@@ -0,0 +1,40 @@
+NIP-53
+======
+
+Nostr-based name registration system
+------------------------------------
+
+`draft` `optional` `author:leandro21b`
+
+## Name:
+
+NostrNameSystem (NNS)
+Kind 50
+
+## Description:
+
+The idea is a nostr-based name registration system, every user can post a name and assign it to a public key, event id, IP address or onion, etc. if other users trust that user who made the registration, they can accept NNS registries from that user and configure your client to search from that user, this can be used for you to find new users on the network without needing the public key, can be used to find an event, can be used as a decentralized DNS, etc. the data is distributed in several relays at the same time and this prevents that, if the user stops updating the NNS list, the data cease to exist, although this theoretically seems to be centralized in a name registration provider user, in practice anyone can provide this service and more than one user can have a copy of the name registration data, the idea is to make it as distributed as possible.
+
+If it happens that 2 people who posted an NNS event have assigned the same name but for different public keys or IP addresses, the client will ask its respective user which one it trusts and the client will start to base itself on the option that the user chooses, then the person who found different keys or IPs can ask the user who published that NNS to correct it.
+
+Another alternative that is already used by some nostr clients is when you follow someone it saves the username automatically and thus displays it when you search for it without having to put the public key of that user every time, this concept can also be applied to domains so that you are not dependent on a centralized DNS server.
+
+Most Internet users do not have a fixed public IP, so the list referring to a website domain may need to be updated from time to time, for this purpose these NNS events will work similarly to kind 3, in case there is an update of an IP related to a domain, the user who posted the NNS event will be able to update the corresponding IP and the most recent event will always be seen by the relays as the correct one and will replace the previous one, as in kind 3.
+
+The owner of a site himself can post a NNS event with a domain and the related IP and if people trust the owner of a site they will be able to accept this NNS event posted by him and so the client and/or the browser will automatically search for that address without the need for a third party to assign names to an IP or public key.
```

Comment:
This all sounds like extending nostr's [pet names](/02.md) to arbitrary data. I would appreciate if the Description  could be boiled down to much fewer lines.

Reply:
More or less, my idea would be a decentralized DNS for urls, I put the possibility of this concept also being applied with usernames and their respective keys as a complement to this, is that one of the objectives with this idea of decentralized DNS based on nostr is to facilitate for whoever runs a relay has a url without depending on a centralized server, the same concept can also be applied to websites, if that is possible with this new NIP-53 I can give a summary and focus more on the idea of decentralized DNS.